pkg/pillar/cmd/upgradeconverter/applydefaultconfigitem.go

```diff
@@ -46,6 +47,11 @@ func applyDefaultConfigItem(ctxPtr *ucContext) error {
 			}
 		}
 	} else {
+		if bootstrapExists {
```

Comment:
Sorry, didn't see this earlier.

I don't know if other microservices will fail if they don't see a ConfigItemValueMap on startup, so I prefer not to change this part.
Can we instead have zedagent know that it should replace it in the case when it has a valid (newer timestamp etc) bootstrap config? That way we always have something in place for other agents hence less ordering dependencies to worry about.

Reply:
The main incentive was to prevent NIM from starting to apply lastresort config, which it does as soon as it sees global config. This has confused customers when they saw diag logs mentioning lastresort even though their override global.json has it disabled.
I checked (and tested) that all microservices start by waiting for global config:
```
	for !ctx.GCInitialized {
		log.Functionf("Waiting for GCInitialized")
		select {
		case change := <-subGlobalConfig.MsgChan():
			subGlobalConfig.ProcessChange(change)
		}
	}
	log.Functionf("processed GlobalConfig")
```
They do not start doing anything until it is available, therefore it is safe to delay generating global config until zedagent processes bootstrap config.

Reply:
OK, but I suspect there are comments about upgradecoverter and/or globalconfig/configitemvaluemap already existing. Such comments would need to be updated. Can you grep/search for such comments?

Reply:
I checked and the only related comments that keep repeating across microservices are:
`Pick up debug aka log level before we start real work`
`Look for global config such as log levels`
I would say they are both still OK.
It seems only zedagent had a mention of global config being guaranteed to exist, but that was removed by this PR:
```
-       // upgradeconverter ensures we have a ConfigItemValueMap so we
-       // read it to get the initial values
```

Reply:
OK